### PR TITLE
Add strategy doc

### PR DESCRIPTION
**Pull request by chengduoZH**

### CODE REVIEW

File: paddle/fluid/pybind/pybind.cc [hunk 1]
```diff
@@ -643,7 +644,16 @@ All parameter, weight, gradient are variables in Paddle.
...
             self.num_threads_ = num_threads;
-          })
+          },
+          R"DOC(The type is INT, num_threads represents the size of thread pool that
+            used to run the operators of the current program in ParallelExecutor.
+            If :math:`num\_threads = 1`, all the operators will execute one by one,
+            but the order maybe difference between iteration.
+            If it is not set, it will be set in ParallelExecutor according to the
+            device count, for GPU, :math:`num\_threads = device\_count * 4`, for CPU,
+            :math:`num\_threads = CPU\_NUM * 4`, the :math:`CPU\_NUM` is an environment
```
Comment (panyx0718) [inline paddle/fluid/pybind/pybind.cc]: clarify that CPU_NUM is an environment var?

File: paddle/fluid/pybind/pybind.cc [hunk 2]
```diff
@@ -678,20 +702,17 @@ All parameter, weight, gradient are variables in Paddle.
     BuildStrategy allows the user to more preciously control how to
     build the SSA Graph in ParallelExecutor by setting the property.
 
-    The available properties include:
```
Comment (panyx0718) [inline paddle/fluid/pybind/pybind.cc]: Not need to remove this?

**Review by panyx0718 — DISMISSED** [review]

File: python/paddle/fluid/parallel_executor.py [hunk 3]
```diff
@@ -31,7 +31,15 @@
...
+    the data across different nodes and every node operates on the data in parallel.
+    If you use ParallelExecutor to run the current program on GPU, the node means GPU
+    device, and ParallelExecutor will get the available GPU device automatically on
+    the current machine. If you use ParallelExecutor to run the current program on CPU,
+    the node means the CPU device, and you can specify the CPU device number by adding
+    'CPU_NUM' environment variable, for example 'CPU_NUM=4', if the environment variable
+    is not found, ParallelExecutor will call `multiprocessing.cpu_count` to get the number
+    of CPUs in the system.
 
     Args:
```
Comment (typhoonzero) [inline python/paddle/fluid/parallel_executor.py:44]: Should add all args docs for parallel executor python api

Comment (chengduoZH) [reply python/paddle/fluid/parallel_executor.py:44]: That is right.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**